src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceInputFormat.java

```diff
@@ -135,4 +132,15 @@ private BulkConnection getBulkConnection(Configuration conf) {
     }
   }
 
+  private RecordReader<Schema, Map<String, ?>> getDelegateRecordReader(String query, Schema schema) {
+    if (SalesforceQueryParser.isRestrictedQuery(query)) {
+      LOG.warn("Restricted SOQL query detected switching to SOAP reader. Performance can be significantly impacted");
```

Comment:
This seems like it would be a confusing message for a user to see. If I saw this I would wonder what a restricted query is and how significant will the performance be impacted.

I think it's better if the message said why it is restricted and describes what exactly that means. For example, something like:

The SOQL query uses an aggregate function. Reads will be performed serially and not in parallel.

This is also expected behavior, so I would make it an info message and not a warning.